domains/client/block-preprocessor/src/inherents.rs

```diff
@@ -87,7 +87,7 @@ where
         .any(|upgraded_runtime_id| upgraded_runtime_id == runtime_id))
 }
 
-fn maybe_runtime_upgrade<CClient, CBlock, Block>(
+pub fn maybe_runtime_upgrade<CClient, CBlock, Block>(
```

Comment:
I think this function could have had a better name and a bit of description. Something like `extract_domain_runtime_upgrade_code` or something. I think this is what it does.

Also even though it wasn't introduced in this PR, I think the function below should first check if there were any runtime upgrades in the header and only then do runtime API call to check if it belongs to this domain. The rationale is that most of the time there are no upgrades and runtime API calls are necessary.

Reply:
there is already a function `has_runtime_upgrade` that checks if there is a runtime upgrade that is used [here](https://github.com/subspace/subspace/blob/main/domains/client/block-preprocessor/src/lib.rs#L173) and this function is used during the during the inherent extrinsic construction and at this point when we need to derive new set_code extrinsic.

Both the calls make one runtime call to fetch the runtimeID of the domain and then bail out if there is no runtime upgrade at given consensus block. If there is a runtime upgrade, then this function ends up extracting it. So I chose to just call this function since the calling the `has_runtime_upgrade` function bring no benefit here. If you have a better approach, I'm open to it as well.

I think i'll also change `has_runtime_upgrade` -> `is_runtime_upgraded`